### PR TITLE
Feature/dp 537 datasharing supplier information dashboard incorrect number of financial information records displayed

### PR DESCRIPTION
**Pull request by maciej-goaco**

### CODE REVIEW

File: Services/CO.CDP.Forms.WebApi/UseCase/GetFormSectionQuestionsUseCase.cs [hunk 1]
```diff
@@ -16,13 +16,18 @@ public class GetFormSectionQuestionsUseCase(IFormRepository formRepository, IMap
         if (section == null)
             return null;
 
-        var answerSet = await formRepository.GetFormAnswerSetsAsync(sectionId, organisationId);
+        var answerSets = await formRepository.GetFormAnswerSetsAsync(sectionId, organisationId);
+        if (answerSets != null)
+        {
+            var currentSharedConsent = answerSets.Select(x => x.SharedConsent).OrderByDescending(x => x.UpdatedOn).FirstOrDefault();
+            answerSets = answerSets.Where(x => x.SharedConsent == currentSharedConsent).ToList();
+        }
```
Comment (jakzal) [inline Services/CO.CDP.Forms.WebApi/UseCase/GetFormSectionQuestionsUseCase.cs:24]: Do we know why the `GetFormAnswerSetsAsync` query does not return what we need?

Comment (maciej-goaco) [reply Services/CO.CDP.Forms.WebApi/UseCase/GetFormSectionQuestionsUseCase.cs]: done.

File: Services/CO.CDP.OrganisationInformation.Persistence/DatabaseFormRepository.cs [hunk 2]
```diff
@@ -81,12 +90,15 @@ public async Task<IEnumerable<FormQuestion>> GetQuestionsAsync(Guid sectionId)
...
-    public async Task<List<FormAnswerSet>> GetFormAnswerSetsAsync(Guid sectionId, Guid organisationId)
+    public async Task<List<FormAnswerSet>> GetFormAnswerSetsFromCurrentSharedConsentAsync(Guid sectionId, Guid organisationId)
     {
-        return await context.Set<FormAnswerSet>()
-            .Include(a => a.Answers)
-            .Include(a => a.SharedConsent)
-            .Where(a => a.Section.Guid == sectionId && a.SharedConsent.Organisation.Guid == organisationId && a.Deleted == false)
+        return await context.Set<SharedConsent>()
+            .Where(x => x.Organisation.Guid == organisationId)
+            .OrderByDescending(x => x.UpdatedOn)
```
Comment (jakzal) [inline Services/CO.CDP.OrganisationInformation.Persistence/DatabaseFormRepository.cs:97]: Use `CreatedOn` in all the cases please. `UpdatedOn` won't point to the latest created shared consent if any information on shared consent is updated on one of the past records.

Comment (maciej-goaco) [reply Services/CO.CDP.OrganisationInformation.Persistence/DatabaseFormRepository.cs]: done

**Review by jakzal — APPROVED** [review]

**Review by dharmverma — APPROVED** [review]

**Pull request merged**